engine/consensus/matching/engine.go

```diff
@@ -31,29 +34,32 @@ import (
-	maxResultsToRequest     int                             // max number of finalized blocks for which we request execution results
-	requireApprovals        bool                            // flag to disable verifying chunk approvals
-	receiptValidator        module.ReceiptValidator         // used to validate receipts
+	unit                   *engine.Unit                    // used to control startup/shutdown
+	log                    zerolog.Logger                  // used to log relevant actions with context
+	engineMetrics          module.EngineMetrics            // used to track sent and received messages
+	tracer                 module.Tracer                   // used to trace execution
+	mempool                module.MempoolMetrics           // used to track mempool size
+	metrics                module.ConsensusMetrics         // used to track consensus metrics
+	state                  protocol.State                  // used to access the  protocol state
+	me                     module.Local                    // used to access local node information
+	receiptRequester       module.Requester                // used to request missing execution receipts by block ID
+	approvalConduit        network.Conduit                 // used to request missing approvals from verification nodes
+	resultsDB              storage.ExecutionResults        // used to check previous results are known
+	headersDB              storage.Headers                 // used to check sealed headers
+	indexDB                storage.Index                   // used to check payloads for results
+	incorporatedResults    mempool.IncorporatedResults     // holds incorporated results in memory
+	receipts               mempool.Receipts                // holds execution receipts in memory
+	approvals              mempool.Approvals               // holds result approvals in memory
+	seals                  mempool.IncorporatedResultSeals // holds the seals that were produced by the matching engine
+	missing                map[flow.Identifier]uint        // track how often a block was missing
+	assigner               module.ChunkAssigner            // chunk assignment object
+	checkingSealing        *atomic.Bool                    // used to rate limit the checksealing call
+	sealingThreshold       uint                            // how many blocks between sealed/finalized before we request execution receipts and approvals
+	maxResultsToRequest    int                             // max number of finalized blocks for which we request execution results
+	requireApprovals       bool                            // flag to disable verifying chunk approvals
+	receiptValidator       module.ReceiptValidator         // used to validate receipts
+	requestTracker         *RequestTracker                 // used to cout the number of approval requests by chunk ([result id][chunk index]=>count)
```

Comment:
The implementation of the `RequestTracker` is _not_ concurrency safe. I think this is critical information which should be included in the variable's goDoc.